app/scripts/controllers/preferences.test.js

```diff
@@ -37,11 +39,14 @@ describe('preferences controller', () => {
       messenger: tokenListMessenger,
     });
 
+    mockEventEmitter = new MockEventEmitter();
+
     preferencesController = new PreferencesController({
       initLangCode: 'en_US',
       tokenListController,
       onInfuraIsBlocked: jest.fn(),
       onInfuraIsUnblocked: jest.fn(),
+      onAccountRemoved: mockEventEmitter.subscribe('accountRemoved'),
```

Comment:
Do you think we can't use a simple mock function here?

Reply:
@mikesposito well the question is what would be the trigger of the function ? how I can make sure it will get triggered ?

Reply:
A good example might be [this one](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/metametrics.test.js#L202)

As it is now it gets triggered when the controller is constructed, which could be a weird behaviour